src/WebJobs.Extensions.DurableTask/Options/DurableTaskOptions.cs

```diff
@@ -35,20 +38,25 @@ public string HubName
         {
             get
             {
-                if (this.hubName == null)
+                if (this.resolvedHubName == null)
                 {
                     // "WEBSITE_SITE_NAME" is an environment variable used in Azure functions infrastructure. When running locally, this can be
                     // specified in local.settings.json file to avoid being defaulted to "TestHubName"
-                    this.hubName = Environment.GetEnvironmentVariable("WEBSITE_SITE_NAME") ?? "TestHubName";
-                    this.defaultHubName = this.hubName;
+                    this.resolvedHubName = Environment.GetEnvironmentVariable("WEBSITE_SITE_NAME") ?? "TestHubName";
+                    this.defaultHubName = this.resolvedHubName;
                 }
 
-                return this.hubName;
+                return this.resolvedHubName;
             }
 
             set
             {
-                this.hubName = value;
+                if (this.originalHubName == null)
```

Comment:
This technically won't work when `SetDefaultHubName()` is called, as it calls `HubName.set`. We could edit that method to set `this.resolvedHubName` directly, so that it doesn't accidentally set `this.originalHubName` in the process.

Reply:
I'm testing this using func.exe and I'm never seeing `SetDefaultHubName()` getting called. In what conditions will it be called?

Reply:
It happens when we need to sanitize the default name, i.e. when the customer's site name has `-` characters. 

Essentially you can test this scenario by setting the Environment variable WEBSITE_SITE_NAME to something like "invalid-site-name" and don't provide a TaskHub.

Reply:
I tested this and was able to successfully trigger the `SetDefaultHubName()` code path and it works in a way that I think is fairly reasonable:

```json
{
  "httpSettings": null,
  "hubName": "Sample-Hub-VS",
  "storageProvider": {
    
  },
  "tracing": {
    "traceInputsAndOutputs": false,
    "traceReplayEvents": false
  },
  "notifications": {
    "eventGrid": null
  },
  "maxConcurrentActivityFunctions": 80,
  "maxConcurrentOrchestratorFunctions": 80,
  "notificationUrl": "http://localhost:7071/runtime/webhooks/durabletask",
  "extendedSessionsEnabled": false,
  "extendedSessionIdleTimeoutInSeconds": 30,
  "maxOrchestrationActions": 100000,
  "overridableExistingInstanceStates": "NonRunningStates",
  "entityMessageReorderWindowInMinutes": 30,
  "useGracefulShutdown": false
}
```

Basically, it set `hubName` to `Sample-Hub-VS`, which was the value I set for `WEBSITE_HOST_NAME`. I guess the alternative could have been that it logs the empty string since nothing was explicitly specified in host.json, though I wonder if this is actually more informative from a debugging perspective. Thoughts?

Reply:
It's probably fine. I have a slight preference for the `""` or `null`, as that is a bit more explicit that the customer didn't provide a `hubName`. That being said, we can easily infer that if the `hubName` matches their site name.

Reply:
I can't think of a simple way to achieve this without further refactoring how we manage hub name. If they don't set it in host.json, the `set` method never gets called until after we do the resolution. At that point this code thinks the resolved value is the original. If you don't mind, I think for the sake of getting this release done we go ahead with what we have here.